### PR TITLE
[NOTIFICATIONS #9] refactor: use new snackbars for wrap/unwrap/approve notifications

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

Migrated onchain transactions notifications to the new solution (`@cowprotocol/snackbars`).

Transactions:
 - approve
 - wrap
 - unwrap
 - cancel
 - eth-flow order placement

Each transaction might end successfully (green snackbar) or with an error (red snackbar).

|    |    |    |    |
|--------|--------|--------|--------|
| image | image | image |     |
| image | image | image | image | 

  # To Test

 - approve (success/error)
 - wrap (success/error)
 - unwrap (success/error)
 - cancel (success/error)
 - eth-flow order placement (success/error)

### CODE REVIEW

**Review by elena-zh — CHANGES_REQUESTED** [review]
Hey @shoom3301  , great job!
However: 
1. When placing a SWAP order in Safe, I see 'view on Safe' link instead of 'View on Explorer'
view on explorer
2. For wraps/unwraps when connected to Safe (WC and inside the Safe), also TXs links should be to Etherscan instead of Safe
safe
See how it is on Prod:

etherscan, not safe

**Comment by shoom3301** [comment]
@elena-zh I can't reproduce the first case.
Do you try a regular swap? Or swap + approve?

image

**Comment by shoom3301** [comment]
About the second case.
I think it's event better. For Safe users we always should lead then to Safe transactions, because in a Safe transaction you can find on-chain transaction but otherwise not. 
@anxolin what do you think?

**Comment by shoom3301** [comment]
@elena-zh the first case is fixed, thank you!

**Comment by elena-zh** [comment]
Hey @shoom3301 , thank you!
There is one more case still not addressed: in Safe (inside), the app allows to place a Buy order with ETH as a sell token (runs through bundling flow). It would be great to have 'View on Explorer' link there  instead of the' View on Safe' on the Submitted toas notification
explorer

Also, when I opened the PR preview link today, tons of notifications appeared on the screen, but the wallet appeared to be disconnected. Can we avoid this situation somehow?  I mean, not to show notifications from the previous day or the ones that were displayed on another device?
image

Thanks

**Review by alfetopito — APPROVED** [review]

**Comment by shoom3301** [comment]
@elena-zh thanks! Fixed the first case.
About the second one, yes, it's possible, but I would like to work on it in the next iteration.

File: apps/cowswap-frontend/src/common/updaters/FinalizeTxUpdater.tsx [hunk 1]
```diff
@@ -124,16 +125,20 @@ function finalizeEthereumTransaction(
...
+        <>{transaction.summary}</>
+      </TransactionContentWithLink>
+    ),
+    id: transaction.hash,
+    icon: isSuccess ? 'success' : 'alert',
+  })
+
+  if (!isSuccess) {
+    getCowSoundError().play()
+  }
```
[the remark this answers is not in the document]
Comment (shoom3301) [reply apps/cowswap-frontend/src/common/updaters/FinalizeTxUpdater.tsx:141]: https://github.com/cowprotocol/cowswap/pull/4039/files#diff-6cc919012b8679e920860edbfee1c224439a6a2aadd41fcce21edce81c446d06L66

It was a part of `soundMiddleware`

**Comment by elena-zh** [comment]
Hey @shoom3301 , great!
But still I have some nitpicks: 
1. When I cancel a TWAP part (not order, opened part), the notification shows 'View on Safe' link instead of the 'View on Etherscan'. Just to mention, 'View on Safe' for cancelling the main order works fine and looks good
2. At lease 2 times I received a Traded 0 amount notification at a time of cancelling a TWAP part (maybe at the time that the order has been filled, but I initiated the Cancel TX and signed it). 
cancelling
3. I placed 2 the same orders, and one of them got stuck in the Opened orders when was filled :( In incognito mode the order goes to History (I assume that the issue is not related to the current PR, so just mentioning) 
stuck
4. I receive 2 Cancelled order notifications, sometimes for Traded as well. I reported #4053 issue for this
5. I have created #4052 issue for case 2 from my [comment ](https://github.com/cowprotocol/cowswap/pull/4038#issuecomment-1999704419)above

Thanks!

**Comment by shoom3301** [comment]
@elena-zh thank you!
1. Fixed
2. This case probably comes out of this task scope, please create another issue for it

**Comment by elena-zh** [comment]
Created #4054 issue for the 2nd case. 
As for the current fix, Order submitted and Order cancelled links for a parent TWAP order now show View on Etherscan link insteat of 'View on Safe. The etherscan link navigates to 404 page
safe

**Comment by elena-zh** [comment]
Hey @shoom3301 , still there are some issues: 
1. Cancel parent TWAP order: the link should say 'View on Safe' (now it navigates to 404 Etherscan). Same should be for order creation pop-up. 
2. Cancel TWAP part: the link should say 'View on Explorer' (now it navigates to Safe). Same should be for Filled and Expired notifications.

**Comment by shoom3301** [comment]
@elena-zh fixed

**Review by elena-zh — APPROVED** [review]
Great job!

**Pull request merged**